src/Microsoft.OData.Edm/Csdl/Semantics/CsdlSemanticsEntityTypeDefinition.cs

```diff
@@ -112,7 +123,18 @@ private IEnumerable<IEdmStructuralProperty> ComputeDeclaredKey()
                 List<IEdmStructuralProperty> key = new List<IEdmStructuralProperty>();
                 foreach (CsdlPropertyReference keyProperty in this.entity.Key.Properties)
                 {
-                    IEdmStructuralProperty structuralProperty = this.FindProperty(keyProperty.PropertyName) as IEdmStructuralProperty;
+                    string propertyName = "";
+                    if (!String.IsNullOrEmpty(keyProperty.Alias))
+                    {
+                        string[] keyProperties = keyProperty.PropertyName.Split('/');
+                        propertyName = keyProperties[1];
+                    }
+                    else 
+                    {
+                        propertyName = keyProperty.PropertyName;
+                    }
+
+                    IEdmStructuralProperty structuralProperty = this.FindProperty(propertyName) as IEdmStructuralProperty;

```

Comment:
If the property is an alias, will it be found by `this.FindProperty()` or will it return null?